rust/nasl-interpreter/src/built_in_functions/misc.rs

```diff
@@ -40,12 +41,67 @@ pub fn dec2str(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue,
     }
 }
 
+/// Compress given data with gzip, when headformat is set to 'gzip' it uses gzipheader.
+pub fn gzip(_: &str, _: &dyn Sink, register: &Register) -> Result<NaslValue, FunctionError> {
+    let data = match register.named("data") {
+        Some(ContextType::Value(NaslValue::String(x))) => x,
+        Some(ContextType::Value(NaslValue::Null)) => return Ok(NaslValue::Null),
+        _ => return Err(FunctionError::new("gzip", ("data", "data").into())),
```

Comment:
in those cases you can also do

```
        x => return Err(FunctionError::new("gzip", ("data", "data", x).into())),
```

This reads like:
`("key", "type", x)` where x is the actual type. 

It is actually handled within the `From<...>` implementations within naslvalue.rs.

While explaining it, it may be better to replace it with another construct for readability sake.